src/fieldAnalyser/typeUtils.ts

```diff
@@ -25,7 +25,22 @@ export function getMostEnglobingType(
   return null;
 }
 
+export function isMultiValueFacet(obj: unknown): boolean {
+  // Handling the most obvious use case. A multi value field could also be a string with delimiter, but we do not want to assume anything...
+  // With an array, we know for sure the metadata has multiple values
+  return Array.isArray(obj);
+}
+
+export function isHierarchicalFacet(obj: unknown): boolean {
```

Comment:
hierarchical facets are dead, long live categoryFacet 😂 
```suggestion
export function isCategoryFacet(obj: unknown): boolean {
```